scripts/spack_packages/packages/geosx/package.py

```diff
@@ -55,41 +50,40 @@ class Geosx(CMakePackage, CudaPackage):
     homepage = "https://github.com/GEOS-DEV/GEOS"
     git = "https://github.com/GEOS-DEV/GEOS.git"
 
-    version('develop', branch='develop', submodules='True')
+    # GEOSX needs submodules to build, but not necessary to build dependencies
+    version('develop', branch='develop')
 
     # SPHINX_BEGIN_VARIANTS
 
     variant('shared', default=True, description='Build Shared Libs.')
     variant('caliper', default=True, description='Build Caliper support.')
-    variant('mkl', default=False, description='Use the Intel MKL library.')
-    variant('essl', default=False, description='Use the IBM ESSL library.')
```

Comment:
Do we not have the virtual blas / lapack dependencies? Or are those only in our current host-configs to be used in building our TPLs (e.g. hypre)?

Reply:
> Do we not have the virtual blas / lapack dependencies? Or are those only in our current host-configs to be used in building our TPLs (e.g. hypre)?

They are specified in the environment `spack.yaml` files like so for blueos:

```
 packages:
    all:
      compiler: [gcc, clang]
      providers:
        mpi: [spectrum-mpi]
        blas: [essl]
        lapack: [essl]

...
    essl:
      buildable: False
      externals:
        - spec: essl@6.3.0.2
          prefix: /usr/tcetmp/packages/essl/essl-6.3.0.2/
```

Reply:
Yeah but we don't directly depend on them? e.g. I don't see `depends("blas")` in our spack recipe. So presumably we only depend on them transitively through e.g. hypre.

Reply:
> Yeah but we don't directly depend on them? I don't see depends("blas") in our spack recipe. So presumably we only depend on them transitively through e.g. hypre.

Ah, we still do (wasn't sure so went and double-checked, as you're right, it could have been pulled in from a difference dependency). There is a "virtual packages" section with mpi/blas/lapack in the original recipe that has remained unchanged: https://github.com/GEOS-DEV/GEOS/blob/00ae74bb80f2a50f536a95036051b5f20592b425/scripts/spack_packages/packages/geosx/package.py#L87-L92